### PR TITLE
MODGOBI-53 Prepend vendorDetail.instructions with vendor code

### PR DESCRIPTION
**Pull request by javalkar-varun**
## Purpose

https://issues.folio.org/browse/MODGOBI-53

## Approach

- In `mapVendorDetail`, refactor logic to prepend vendor code to instructions if it exists else map vendor instructions to vendor code

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/gobi/Mapper.java [hunk 1]
```diff
@@ -651,14 +652,22 @@ private void mapLocation(List<CompletableFuture<?>> futures, Location location,
...
     Optional.ofNullable(mappings.get(Mapping.Field.NOTE_FROM_VENDOR))
+    .ifPresent(field -> futures.add(field.resolve(doc)
+      .thenAccept(o -> vendorDetail.setNoteFromVendor((String) o))
+      .exceptionally(Mapper::logException)));
+    
+    Optional.ofNullable(mappings.get(Mapping.Field.VENDOR_INSTRUCTIONS))
       .ifPresent(field -> futures.add(field.resolve(doc)
-        .thenAccept(o -> vendorDetail.setNoteFromVendor((String) o))
+        .thenAccept(o -> {
+        	String vendorCode = vendorDetail.getNoteFromVendor();
```
Comment (KVupp) [inline src/main/java/org/folio/gobi/Mapper.java]: As these are futures, there is no guarantee that the NOTE_FROM_VENDOR code above is always executed before this line. This code must move inside the NOTE_FROM_VENDOR block if we have to set this field conditionally.

Comment (javalkar-varun) [reply src/main/java/org/folio/gobi/Mapper.java]: yeah... Earlier, when I pushed changes, Jenkins build fine but my local build kept failing randomly. I was wondering something was missing until your comment. I have refactored the code to move it inside `NOTE_FROM_VENDOR`. Looks good now. Thanks for catching it!

Comment (KVupp) [reply src/main/java/org/folio/gobi/Mapper.java]: is vendor_instructions supposed to be set only if NOTE_FROM_VENDOR? I just looked at the JIRA bug and doesn't look like it. Questions here is In the existing code Vendor code is already set to NOTE_FROM_VENDOR. If it is already present in the field, should it also go into another field VENDOR_INSTRUCTIONS?
If yes, then this code needs to be changed to use combinator. The way it is changed in line 661, the //orderNotes will be dropped if //purchaseOption/vendorCode is not present

Comment (javalkar-varun) [reply src/main/java/org/folio/gobi/Mapper.java]: If //purchaseOption/vendorCode is not present, we want to keep //orderNotes as is. Quote from jira; I think its implied here `If there is already data in the Instructions to Vendor field (which would have come from the GobiAPI OrderNotes field), place the 5 character code before that data, followed by a space, e.g. EBS3U Please rush.`

Comment (javalkar-varun) [reply src/main/java/org/folio/gobi/Mapper.java]: BTW, from what I understand, //purchaseOption/vendorCode will always be there

Comment (KVupp) [reply src/main/java/org/folio/gobi/Mapper.java]: > If //purchaseOption/vendorCode is not present, we want to keep //orderNotes as is. Quote from jira; I think its implied here `If there is already data in the Instructions to Vendor field (which would have come from the GobiAPI OrderNotes field), place the 5 character code before that data, followed by a space, e.g. EBS3U Please rush.`

correct we want to keep the //orderNotes irrespective of /vendorcode. The code here doesn't fo that. If //orderNotes is not present/Null. (which is mapped to NOTE_FROM_VENDOR). It doesn't go into the inner block and thus making VENDOR_INSTRUCTIONS null.

Comment (KVupp) [reply src/main/java/org/folio/gobi/Mapper.java]: > BTW, from what I understand, //purchaseOption/vendorCode will always be there

I don't think it's a mandatory field, we have few examples (PO_listed_print_monograph) where the //purchaseOption/vendorCode field doesn't come through.

**Comment by javalkar-varun** [comment]
Update from standup discussion 3/14:
mod-gobi already has a mapping for `//PurchaseOption/VendorCode` https://github.com/folio-org/mod-gobi/blob/master/src/main/resources/ListedElectronicMonograph.json#L133
We can use existing mapping as is instead of creating a new one. As a result, closing this PR.

**Pull request closed without merging**